cli/tools/registry/graph.rs

```diff
@@ -12,137 +16,189 @@ use deno_semver::jsr::JsrPackageReqReference;
+                  let maybe_version = graph
+                    .get(&resolution.specifier)
+                    .and_then(|m| m.npm())
+                    .map(|n| n.nv_reference.nv().version.clone());
+                  diagnostics_collector.push(
+                    PublishDiagnostic::MissingConstraint {
+                      specifier: resolution.specifier.clone(),
+                      specifier_text: specifier_text.to_string(),
+                      resolved_version: maybe_version,
+                      text_info: SourceTextInfo::new(source_text.clone()),
+                      referrer: resolution.range.clone(),
+                    },
+                  );
+                }
               }
             }
-          }
-          "http" | "https" => {
-            skip_specifiers.insert(resolution.specifier.clone());
-            diagnostics_collector.push(
-              PublishDiagnostic::InvalidExternalImport {
-                kind: format!("non-JSR '{}'", resolution.specifier.scheme()),
-                text_info: SourceTextInfo::new(source_text.clone()),
-                imported: resolution.specifier.clone(),
-                referrer: resolution.range.clone(),
-              },
-            );
-          }
-          _ => {
-            skip_specifiers.insert(resolution.specifier.clone());
-            diagnostics_collector.push(
-              PublishDiagnostic::InvalidExternalImport {
-                kind: format!("'{}'", resolution.specifier.scheme()),
-                text_info: SourceTextInfo::new(source_text.clone()),
-                imported: resolution.specifier.clone(),
-                referrer: resolution.range.clone(),
-              },
-            );
+            "http" | "https" => {
+              skip_specifiers.insert(resolution.specifier.clone());
+              diagnostics_collector.push(
+                PublishDiagnostic::InvalidExternalImport {
+                  kind: format!("non-JSR '{}'", resolution.specifier.scheme()),
+                  text_info: SourceTextInfo::new(source_text.clone()),
+                  imported: resolution.specifier.clone(),
+                  referrer: resolution.range.clone(),
+                },
+              );
+            }
+            _ => {
+              skip_specifiers.insert(resolution.specifier.clone());
+              diagnostics_collector.push(
+                PublishDiagnostic::InvalidExternalImport {
+                  kind: format!("'{}'", resolution.specifier.scheme()),
+                  text_info: SourceTextInfo::new(source_text.clone()),
+                  imported: resolution.specifier.clone(),
+                  referrer: resolution.range.clone(),
+                },
+              );
+            }
           }
         }
-      }
+      };
+
+    let options = WalkOptions {
+      check_js: true,
+      follow_dynamic: true,
+      // this being disabled will cause it to follow everything in the graph
```

Comment:
Is this bit important here? If so I would add a highlight

Reply:
What do you mean by "add a highlight"?

Reply:
Something like `IMPORTANT: this being disabled ...`. IIRC this is actually significant because previously we didn't raise diagnostics for all files just exports.